sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from "../util/logging";
+
+import * as child_process from "child_process";
+
+/**
+ * Provides the user access token and expire time
+ * with Azure CLI command "az account get-access-token".
+ */
+export class AzureCliCredential implements TokenCredential {
+  /**
+   * Creates an instance of the AzureCliCredential class.
+   */
+  constructor() {}
+
+  /**
+   * Gets the access token from Azure CLI
+   * @param resource The resource to use when getting the token
+   */
+  protected async getAzureCliAccessToken(resource: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        child_process.exec(
+          `az account get-access-token --output json --resource ${resource}`,
+          (error, stdout, stderr) => {
+            resolve({ stdout: stdout, stderr: stderr });
+          }
+        );
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  /**
+   * Authenticates with Azure Active Directory and returns an access token if
+   * successful.  If authentication cannot be performed at this time, this method may
+   * return null.  If an error occurs during authentication, an {@link AuthenticationError}
+   * containing failure details will be thrown.
+   *
+   * @param scopes The list of scopes for which the token will have access.
+   * @param options The options used to configure any requests this
+   *                TokenCredential implementation might make.
+   */
+  public async getToken(
+    scopes: string | string[],
+    options?: GetTokenOptions
+  ): Promise<AccessToken | null> {
+    return new Promise((resolve, reject) => {
+      let scope: string;
+      scope = typeof scopes === "string" ? scopes : scopes[0];
+      logger.info(`use the scope ${scope}`);
+      const resource = scope.replace(/\/.default$/, "");
+      let responseData = "";
+
+      const { span } = createSpan("AzureCliCredential-getToken", options);
+      this.getAzureCliAccessToken(resource)
+        .then((obj: any) => {
+          if (obj.stderr) {
+            let isLoginError = obj.stderr.match("Please run 'az login' from a command prompt to authenticate before using this credential.");
```

Comment:
I don't know this API, so I have to ask: is this checking whether stderr contains that literal?

Reply:
Looks like it's a regex match.  I might just change it to look for the 'az login' part.

Reply:
Updated